src/MobileRedirection.php

```diff
@@ -64,7 +64,16 @@ public function register() {
 		add_filter( 'amp_default_options', [ $this, 'filter_default_options' ] );
 		add_filter( 'amp_options_updating', [ $this, 'sanitize_options' ], 10, 2 );
 
-		if ( AMP_Options_Manager::get_option( Option::MOBILE_REDIRECT ) && ! amp_is_canonical() ) {
+		$is_amp_canonical           = amp_is_canonical();
+		$sandboxing_level           = amp_get_sandboxing_level();
+		$is_mobile_redirect_enabled = AMP_Options_Manager::get_option( Option::MOBILE_REDIRECT );
+
+		// Add alternative link if mobile redirection is enabled or sandboxing level is set to loose or moderate.
+		if ( ! $is_amp_canonical && ( $is_mobile_redirect_enabled || ( 1 === $sandboxing_level || 2 === $sandboxing_level ) ) ) {
+			add_action( 'template_redirect', [ $this, 'add_mobile_alternative_link_head_hook' ], PHP_INT_MAX );
```

Comment:
I saw your comment in https://github.com/ampproject/amp-wp/pull/7288#discussion_r1028929720 but I don't quite understand still why `add_mobile_alternative_link_head_hook` is needed. The `amp_is_available()` function will complain if it is called before `template_redirect` (or rather `wp`), but `wp_head` runs way after both. So to me it doesn't seem that `add_mobile_alternative_link_head_hook` is needed and we can just do:

```suggestion
			add_action( 'wp_head', [ $this, 'add_mobile_alternative_link' ] );
```

Reply:
Sorry for the confusion here. Actually, this confusion arises from tests where we are directly calling `add_mobile_alternative_link` to get the alternate link string in tests, and since we are checking `amp_is_request` in the function it was throwing an error. I have fixed it now.